### ISSUE TITLE
Image not shown inside Hero after navigating to ListView in Flutter

### ISSUE THREAD

**Issue by hadleycode** [flutter/flutter#43605]
The original image is not shown inside a Hero widget after navigating to a screen with a ListView and scrolling to the bottom of the list and then navigating back.

You can see the problem visually here -> https://imgur.com/a/nXl26yX

To reproduce: 
1. Page A has images (each inside a Hero)
2. Navigate to page B which is a ListView (each image in the ListView is also inside a Hero tag)
3. Scroll to the bottom of the list
4. Return to page A 
5. The original images are no longer visible on page A. 

I have the impression this is because the ListView will delete the first cells when they scroll out of the screen. And therefore the Hero widgets which had references to the tags on page A are also deleted, thus creating the bug of the images not appearing when returning to screen A. Should the reference to the widget on screen A not be handled by the Flutter framework? I have already tried modifying my widget tree in different ways, but with no success yet. 

Full code to reproduce can be found here -> https://github.com/hadleycode/flutter_hero_bug

This happens on emulators, simulators and physical phones - both Android and iOS.

```
[✓] Flutter (Channel stable, v1.9.1+hotfix.4, on Mac OS X 10.14.6 18G95, locale fr-FR)
    • Flutter version 1.9.1+hotfix.4 at /Users/hadley/Development/flutterSDK/flutter
    • Framework revision cc949a8e8b (4 weeks ago), 2019-09-27 15:04:59 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/hadley/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.1, Build version 11A1027
    • CocoaPods version 1.8.0

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

```

**Comment by MostafaKhalile** [comment]
did you add the same tag in both of hero widgets in Pages (A&B) 
### Like That 
**`child: Hero(
                        tag: 'logo',
                        child: Image.asset(
                          'assets/logo.png',
                          color: Colors.white,
                          height: 250.0,
                          width: 250.0,
                        ),
                      ),`**

**Comment by hadleycode** [comment]
This issue seems to be fixed by V1.9.1 hotfix 

https://github.com/flutter/flutter/commit/68587a0916366e9512a78df22c44163d041dd5f3
